recipes/functorch/meta.yaml

```
@@ -0,0 +1,56 @@
+{% set name = "functorch" %}
+{% set version = "0.1.1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/pytorch/functorch/archive/v{{ version }}.tar.gz
+  sha256: 3b297f4acf1d61fd95b86bd2ec1805bc9c236bb0fb341a42c2300a4b3d497fbd
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  skip: true  # [win]
+  detect_binary_files_with_prefix: false
+  missing_dso_whitelist:
+    # conda-forge::pytorch provides these libs
+    - "$RPATH/libc10_cuda.so"
+    - "$RPATH/libtorch_python.so"
+    - "$RPATH/libtorch_cpu.so"
+    - "$RPATH/libc10.so"
+    - "$RPATH/libc10_cuda.dylib"
+    - "$RPATH/libtorch_python.dylib"
+    - "$RPATH/libtorch_cpu.dylib"
+    - "$RPATH/libc10.dylib"
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - pip
+    - python
+    - pytorch
```

```suggestion
    - pytorch >=1.10.0,<2
```
There is a minimum pytorch version requirement.

https://github.com/pytorch/functorch/blob/9b96f14e65ffdf64a28416054fe536ed14297fe9/setup.py#L48

---

I don't think this is correct. If it does link to the pytorch ABI the migrator will skip this. I think you can leave it in a comment, but the pinnings will take care of this for you.

---

I expected the global pinning a specified pinning to be combined. i.e.

```
pytorch >=1.10,<2,1.11
```

Where 1.11 is the global pinning for pytorch. Because you can set a python version range on a package and the incompatible python builds are skipped? Are you saying the migrator won't migrate this recipe to pytorch =1.12 or that it won't migrate to pytorch=2.0 because I added the major version requirement?

---

Lets see what happens. https://github.com/conda-forge/torchvision-feedstock/pull/56

---

last i remember, the migrator looks for pinnings that aren't set. It will provide a pinning for those. I had to go through some gymnasitcs where a requirement is listed twice, once without any constraints, the other with the constraints.

---

https://github.com/conda-forge/torchvision-feedstock/pull/56#pullrequestreview-1033456278

---

to be specific, i don't think it will migrate to 1.12 AND it will ignore the pin of 1.11.

The problem is that it might solve, at compile time, to pytorch 1.10, and build a version of the package that is incompatible with the rest of the ecosystem.

---

This is good to know. Thanks for taking the time to research that for me @hmaarrfk. 😄 

This is a problem because if you don't check the global pinning for every dependency, you could be overriding it without knowing!

---

Yes. This is true.